accepted/2022/simplify-output-paths.md

```diff
@@ -0,0 +1,53 @@
+# Simplify .NET Output Paths
+
+We'd like to simplify the output paths for .NET 8.
+
+Currently, the default output path for a .NET project includes folders for the `TargetFramework` and the `RuntimeIdentifier` (if it is set).  Additionally, the publish output goes in a `publish` subfolder of the output folder.  The resulting paths are as follows:
+
+- `bin\<Configuration>\<TargetFramework>` - Build output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>` - Build output with `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\publish` - Publish output with no `RuntimeIdentifier`
+- `bin\<Configuration>\<TargetFramework>\<RuntimeIdentifier>\publish` - Publish output with `RuntimeIdentifier`
+
+This makes it harder to find the output of a project, and sometimes nests one type of output in another output folder.
+
+## Proposed behavior
+
+Projects targeting .NET 8 and higher will by default not include the `TargetFramework` in the output path unless the project is multi-targeted.  Similarly, the `RuntimeIdentifier` will not by default be part of the output path.  Finally, the publish output path will be `bin\publish` by default instead of a `publish` subfolder inside of the output path.
+
+These would be the default output paths for .NET 8 projects:
+
+- `bin\<Configuration>` - Build output path for single targeted project
+- `bin\<Configuration>\<TargetFramework>` - Build output for multi-targeted project
+- `bin\publish` - Publish output for single targeted project
+- `bin\publish\<TargetFramework>` - Publish output for multi-targeted project
```

Comment:
Having `dotnet build` use a `Configuration` sub-folder while `dotnet publish` doesn't seems like a bizarre inconsistency to me. I have to imagine many will find that confusing.

I almost feel like a new `pub\<Configuration>[\<TargetFramework>]` scheme would make more sense than this. It would mean a `.gitignore` update, but it would achieve the goal of avoiding deeply nested output folders, and it would more clearly separate publish output from build output.

Reply:
I agree, another point is that these will overwrite each other: `dotnet publish -c MyRelease` and `dotnet publish -c YourRelease` - I believe it's pretty well known that if you want two independent builds, you use two different configurations, but that won't work anymore.

Reply:
I've clarified that the publish path would only change for the `Release` configuration, and added a section explaining the motivation behind putting publish at the same level as Configuration.